src/ocf_lru.c

```diff
@@ -810,19 +796,18 @@ void ocf_lru_hot_cline(ocf_cache_t cache, ocf_cache_line_t cline)
 	if (hot)
 		return;
 
-	list = lru_get_cline_list(cache, cline);
+	part_id = ocf_metadata_get_partition_id(cache, cline);
+	part = &cache->user_parts[part_id].part;
+	clean = !metadata_test_dirty(cache, cline);
+	list = ocf_lru_get_list(part, lru_list, clean);
 
 	OCF_METADATA_LRU_WR_LOCK(cline);
 
-	if (node->next != end_marker ||
-			node->prev != end_marker ||
-			list->head == cline || list->tail == cline) {
-		remove_lru_list(cache, list, cline);
-	}
+	/* cacheline must be on the list when set_hot gets called */
+	ENV_BUG_ON(node->next == end_marker && list->tail != cline);
+	ENV_BUG_ON(node->next == end_marker && list->tail != cline);
```

Comment:
Was this supposed to be a different bugon?

Reply:
good catch - fixed